### PR TITLE
Replaced GMock with Trompeloeil

### PR DESCRIPTION
**Pull request by slci**
- Simplified test_runner with Catch2 + Trompeloeil
- Reformatted sources with clang-format-11
- Added example Trompeloeil test case

### CODE REVIEW

File: tests/test_runner/main.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        else
+        {
+            CAPTURE(failure);
+            CHECK(failure.empty());
+        }
+    });
+
+    return Catch::Session().run(argc, argv);
+}
```
Comment (insooth) [inline tests/test_runner/main.cpp:31]: Please add a new line here.

Comment (slci) [reply tests/test_runner/main.cpp:31]: I guess clang-format shall do it automatically basing on the configuration we have. Will fix.

Comment (slci) [reply tests/test_runner/main.cpp]: Done, but unfortunatelly clang-format won't add new lines at EOFs automatically.

File: .clang-format [hunk 2]
```diff
@@ -69,7 +68,7 @@ BraceWrapping:
   AfterStruct:     true
   AfterUnion:      false
   BeforeCatch:     true
-  BeforeElse:      false
+  BeforeElse:      true
 # shall work in clang-format-10, but does not...
 #  BeforeLambdaBody: false
```
Comment (bburda) [inline .clang-format:73]: Does it work with clang-format-11?

File: tests/lib1_test.cpp [hunk 3]
```diff
@@ -1,13 +1,42 @@
...
+    acc += i.Function("One");
+    acc += i.Function("Three");
+    acc += i.Function("Two");
+
+    return acc;
+}
+
 TEST_CASE("Answer is given", "[answer]")
 {
     // REQUIRE(library1::answer() == 123);
```
Comment (bburda) [inline tests/lib1_test.cpp:24]: Can we remove it?

File: .clang-format (hunk 2 above)
Comment (slci) [reply .clang-format:73]: It does.

File: tests/lib1_test.cpp (hunk 3 above)
Comment (slci) [reply tests/lib1_test.cpp:24]: Sure

**Pull request merged**